mantle/kola/tests/misc/nfs-client.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2025 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package misc
+
+import (
+	"fmt"
+	"path"
+	"time"
+
+	"github.com/coreos/coreos-assembler/mantle/kola"
+	"github.com/coreos/coreos-assembler/mantle/kola/cluster"
+	"github.com/coreos/coreos-assembler/mantle/kola/register"
+	"github.com/coreos/coreos-assembler/mantle/platform/conf"
+	"github.com/coreos/coreos-assembler/mantle/util"
+)
+
+const (
+machine_conf = `
+variant: fcos
+version: 1.5.0
+storage:
+  files:
+    - path: /etc/containers/systemd/nfs.container
+      overwrite: true
+      contents:
+        inline: |
+          [Container]
+          Image=quay.io/coreos-assembler/nfs
+          Volume=/tmp:/export
+          Network=host
+          PodmanArgs=--privileged
+          [Install]
+          WantedBy=default.target
+    - path: "/etc/hostname"
+      contents:
+        inline: "nfs-client"
+      mode: 0644
+systemd:
+  units:
+    - name: "var-mnt-nfsv4.mount"
+      enabled: true
+      contents: |-
+        [Unit]
+        Description=NFS Client
+        After=network-online.target
+        Requires=network-online.target
+        After=rpc-statd.service nfs.service
+        Requires=rpc-statd.service
+
+        [Mount]
+        What=127.0.0.1:/
+        Where=/var/mnt/nfsv4
+        Type=nfs4
+        Options=defaults,noexec,nfsvers=4
+
+        [Install]
+        WantedBy=multi-user.target
+    - name: "var-mnt-nfs.mount"
+      enabled: true
+      contents: |-
+        [Unit]
+        Description=NFS Client
+        After=network-online.target
+        Requires=network-online.target
+        After=rpc-statd.service nfs.service
+        Requires=rpc-statd.service
+
+        [Mount]
+        What=127.0.0.1:/export
+        Where=/var/mnt/nfs
+        Type=nfs
+        Options=vers=3
+
+        [Install]
+        WantedBy=multi-user.target`
+)
+
+// Test nfs client
+
+func init() {
+	register.RegisterTest(&register.Test{
+		Run:         nfsClientTest,
+		ClusterSize: 1,
+		UserData:    conf.Butane(machine_conf),
+		Name:        "linux.nfs.client",
+		Description: "Verifies NFS client works.",
+		Tags:        []string{kola.NeedsInternetTag},
+		Platforms:   []string{"qemu"},
+
+		// RHCOS has a separate test for NFS v4 server and client
+		ExcludeDistros: []string{"rhcos"},
```

Comment:
Is it needed to run on rhcos for this test?

Reply:
rhcos has another test from [this file](mantle/kola/tests/rhcos/nfs.go) that test the server (v4) too. 
In the new test added here, we test only the nfs client and is specific to fcos. Am I missing other distros to be excluded here?

Reply:
I am OK to just exclude `rhcos`, and we have the third distro `scos`, it will test both.